flow/project.py

```diff
@@ -826,9 +826,16 @@ def _map_scheduler_jobs(scheduler_jobs):
                 expanded = JobOperation.expand_id(name)
                 yield expanded['job_id'], expanded['operation-name'], sjob
 
-    def _get_operations_status(self, job):
+    def _get_operations_status(self, job, cached_status=None):
```

Comment:
I don't think it's necessary to pass this argument along here. We can always assume that the *current* status is represented by `self.document._status`.

Reply:
The `cached_status = self.document.get('_status', dict())` in this function is actually the most expensive part, so passing it through is the main point of the optimization.